target/linux/imx6/image/Makefile

```diff
@@ -235,4 +235,38 @@ define Device/apalis
 endef
 TARGET_DEVICES += apalis
 
+define Device/xiaomi_gateway-lumi
+  DEVICE_VENDOR := Xiaomi
+  DEVICE_MODEL := Lumi Gateway
+  DEVICE_PACKAGES := \
+	kmod-button-hotplug kmod-input-gpio-keys \
+	kmod-ledtrig-activity kmod-ledtrig-oneshot \
+	kmod-ledtrig-transient kmod-ledtrig-gpio \
+	kmod-i2c-core kmod-iio-core kmod-iio-vf610 \
+	kmod-sound-core kmod-sound-soc-imx \
+	kmod-sound-soc-tfa9882 alsa-utils \
+	wpa-supplicant ca-certificates hostapd \
+	nand-utils kobs-ng
+
+  KERNEL_INSTALL := 0
+  KERNEL_NAME := zImage
+  KERNEL_SUFFIX := -zImage
+  KERNEL := kernel-bin
+  IMAGES := sysupgrade.bin dtb rootfs.bin
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  IMAGE/rootfs.bin := append-rootfs
+  IMAGE/dtb := install-dtb
+endef
+
+define Device/xiaomi_dgnwg05lm
+  $(Device/xiaomi_gateway-lumi)
+  DEVICE_MODEL := DGNWG05LM
+  BOARD_NAME := xiaomi,dgnwg05lm
```

Comment:
BOARD_NAME should not be needed anymore.

Reply:
As I said https://github.com/openwrt/openwrt/pull/4053#discussion_r615370393
it is the only way to pass the check for nand upgrade. Otherwise, it keeps underscore in the directory path inside sysupgrade binary instead of a comma. https://github.com/openwrt/openwrt/blob/master/package/base-files/files/lib/upgrade/nand.sh#L347-L353

Reply:
It's "the only way" just because you think it's impossible to replace a comma by an underscore? Please, try to implement a wildcard-based approach so we get rid of these issues...

Reply:
Removed BOARD_NAME in favor of more tricky `platform_check_image()`